app/src/main/java/org/schabi/newpipe/SponsorBlockApiTask.java

```diff
@@ -0,0 +1,174 @@
+package org.schabi.newpipe;
+
+import android.annotation.SuppressLint;
+import android.app.Application;
+import android.content.Context;
+import android.net.ConnectivityManager;
+import android.os.AsyncTask;
+import android.util.Log;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.schabi.newpipe.util.SponsorTimeInfo;
+import org.schabi.newpipe.util.TimeFrame;
+
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+import java.util.Random;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+
+public class SponsorBlockApiTask extends AsyncTask<String, Void, JSONObject> {
+    private static final Application app = App.getApp();
+    private static final String sponsorBlockApiUrl = "https://api.sponsor.ajay.app/api/";
+    private static final int timeoutPeriod = 30;
+    private static final String TAG = SponsorBlockApiTask.class.getSimpleName();
+    private static final boolean DEBUG = MainActivity.DEBUG;
+    private OkHttpClient client;
+
+    // api methods
+    public SponsorTimeInfo getVideoSponsorTimes(String url) throws ExecutionException, InterruptedException, JSONException {
+        String videoId = parseIdFromUrl(url);
```

Comment:
This should only be done when the video is a YouTube video. However, please take a look at our  extractor. Getting a videoId from an url is already implemented there.

Reply:
done, now checks if info.getUrl().startsWith("https://www.youtube.com")